src/WebJobs.Script/Description/DocumentDBBindingMetadata.cs

```diff
@@ -27,7 +26,12 @@ public override void ApplyToConfig(JobHostConfigurationBuilder configBuilder)
             DocumentDBConfiguration config = new DocumentDBConfiguration();
             if (!string.IsNullOrEmpty(ConnectionString))
             {
-                config.ConnectionString = ConnectionString;
+                // Directly resolve this. Ideally this would be handled with 
+                // [AllowNameResolution]. We can do that when the following
+                // issue is resolved:
+                // https://github.com/projectkudu/WebJobsPortal/issues/117
```

Comment:
As I mentioned on this bug, I'm now working on multi-storage account support in the script layer. That support (and the core SDK) expect that the value that the portal gives is an **app setting name**, pointing to the connection string. I.e. I'll be taking the setting value that the portal gives me and plumbing it directly down into the existing StorageAccountAttribute which expects a setting name (not a full connection string). If there are any %% on the portal value, it would mean a level of indirection - that the NameResolver would resolve the value, which should still be a connection/app setting name, not the full connection string. I think the portal is doing the right thing.

If the DocDB binding needs a connection string, it should be the one to take the resolved setting name (which is a pointer to the connection string) and resolve it further.

I'll finish up the multi-storage account work and get it out for review soon, so you guys can see.